File: autocare-frontend/src/services/customerService.js
import api from "./api";

const customerService = {

    getAllCustomers(page = 0, size = 5) {
        return api.get(`/customers?page=${page}&size=${size}`);
    },

    getCustomerById(id) {
        return api.get(`/customers/${id}`);
    },

    createCustomer(data) {
        return api.post("/customers", data);
    },

    updateCustomer(id, data) {
        return api.put(`/customers/${id}`, data);
    },

    deleteCustomer(id) {
        return api.delete(`/customers/${id}`);
    }

};

export default customerService;

File: autocare-backend/src/main/java/com/pobitra/autocare/service/CustomerService.java
package com.pobitra.autocare.service;

import com.pobitra.autocare.dto.CustomerRequestDTO;
import com.pobitra.autocare.dto.CustomerResponseDTO;
import com.pobitra.autocare.dto.PageResponseDTO;
import com.pobitra.autocare.entity.Customer;

import java.util.List;
import java.util.Optional;
import org.springframework.data.domain.Page;

public interface CustomerService {

    CustomerResponseDTO saveCustomer(CustomerRequestDTO customerRequestDTO);



    PageResponseDTO<CustomerResponseDTO> getAllCustomers(
            int page,
            int size,
            String sortBy,
            String direction
    );

    Customer getCustomerById(Long id);

    Customer updateCustomer(Long id, Customer customer);

    void deleteCustomer(Long id);
}